group-coordinator/src/main/java/org/apache/kafka/coordinator/group/ShareGroupAutoOffsetResetStrategy.java

```diff
@@ -0,0 +1,162 @@
+        this.duration = Optional.of(duration);
+    }
+
+    /**
+     *  Returns the AutoOffsetResetStrategy from the given string.
+     */
+    public static ShareGroupAutoOffsetResetStrategy fromString(String offsetStrategy) {
+        if (offsetStrategy == null) {
+            throw new IllegalArgumentException("Auto offset reset strategy is null");
+        }
+
+        if (StrategyType.BY_DURATION.toString().equals(offsetStrategy)) {
+            throw new IllegalArgumentException("<:duration> part is missing in by_duration auto offset reset strategy.");
+        }
+
+        if (Arrays.asList(Utils.enumOptions(StrategyType.class)).contains(offsetStrategy)) {
+            StrategyType type = StrategyType.valueOf(offsetStrategy.toUpperCase(Locale.ROOT));
+            switch (type) {
+                case EARLIEST:
+                    return EARLIEST;
+                case LATEST:
+                    return LATEST;
+                default:
+                    throw new IllegalArgumentException("Unknown auto offset reset strategy: " + offsetStrategy);
+            }
+        }
+
+        if (offsetStrategy.startsWith(StrategyType.BY_DURATION + ":")) {
+            String isoDuration = offsetStrategy.substring(StrategyType.BY_DURATION.toString().length() + 1);
+            try {
+                Duration duration = Duration.parse(isoDuration);
+                if (duration.isNegative()) {
+                    throw new IllegalArgumentException("Negative duration is not supported in by_duration offset reset strategy.");
+                }
+                return new ShareGroupAutoOffsetResetStrategy(duration);
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Unable to parse duration string in by_duration offset reset strategy.", e);
+            }
+        }
+
+        throw new IllegalArgumentException("Unknown auto offset reset strategy: " + offsetStrategy);
+    }
+
+    /**
+     * Returns the offset reset strategy type.
+     */
+    public StrategyType type() {
+        return type;
+    }
+
+    /**
+     * Returns the name of the offset reset strategy.
+     */
+    public String name() {
+        return type.toString();
+    }
+
+    /**
+     * Return the timestamp to be used for the ListOffsetsRequest.
+     * @return the timestamp for the OffsetResetStrategy,
+     * if the strategy is EARLIEST or LATEST or duration is provided
+     * else return Optional.empty()
+     */
+    public Optional<Long> timestamp() {
+        if (type == StrategyType.EARLIEST)
+            return Optional.of(ListOffsetsRequest.EARLIEST_TIMESTAMP);
+        else if (type == StrategyType.LATEST)
+            return Optional.of(ListOffsetsRequest.LATEST_TIMESTAMP);
+        else if (type == StrategyType.BY_DURATION && duration.isPresent()) {
+            Instant now = Instant.now();
+            return Optional.of(now.minus(duration.get()).toEpochMilli());
+        } else
+            return Optional.empty();
```

Comment:
Would it be better to use `pattern type` switch statement, java 17, so any change in enum and not in this method can give compile time exception?